src/sql/workbench/parts/notebook/common/models/cell.ts

```diff
@@ -64,6 +68,8 @@ export class CellModel implements ICellModel {
 		} else {
 			this._isTrusted = false;
 		}
+		// if the fromJson() method was already called and _cellGuid was previously set, don't generate another UUID unnecessarily
+		this._cellGuid = this._cellGuid ? this.cellGuid : generateUuid();
```

Comment:
a) at a minimum should consistently use _cellGuid (you use public one once)
b) can we use `this._callGuid = this._cellGuid || generateUuid();` instead

Reply:
Whoops nice catch. Fixed.